### PR TITLE
EFA-418 Add progress dialog for account deletion

### PR DESCRIPTION
**Pull request by ig-garcia**

### CODE REVIEW

File: k9mail/src/main/java/security/planck/ui/removeaccount/RemoveAccountViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                    removeAccount()
+                }
+
+            is RemoveAccountState.Done ->
+                stateLiveData.value = RemoveAccountState.Finish(true)
+
+            is RemoveAccountState.AccountNotAvailable ->
+                stateLiveData.value = RemoveAccountState.Finish(true)
+
+            else -> error("unexpected state: ${stateLiveData.value}")
```
Comment (Stereodark) [inline k9mail/src/main/java/security/planck/ui/removeaccount/RemoveAccountViewModel.kt:59]: I think we can write just Account not available here

File: k9mail/src/main/java/security/planck/ui/removeaccount/RemoveAccountViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        }
+    }
+
+    fun negativeAction() {
+        when (stateLiveData.value) {
+            is RemoveAccountState.RemoveAccountConfirmation ->
+                stateLiveData.value = RemoveAccountState.Finish(false)
+
+            else -> error("unexpected state: ${stateLiveData.value}")
```
Comment (Stereodark) [inline k9mail/src/main/java/security/planck/ui/removeaccount/RemoveAccountViewModel.kt:68]: and here something like user cancelled

**Review by Stereodark — APPROVED** [review]
Looks nice, very good refactoring as well

File: k9mail/src/main/java/security/planck/ui/removeaccount/RemoveAccountViewModel.kt (hunk 1 above)
Comment (ig-garcia) [reply k9mail/src/main/java/security/planck/ui/removeaccount/RemoveAccountViewModel.kt:59]: 'when' is actually like Java's 'switch', and 'else' is like Java's 'default'. 'error' throws an IllegalStateException. So this is more of an error for the developer, we only expect to have this method called in these states.

File: k9mail/src/main/java/security/planck/ui/removeaccount/RemoveAccountViewModel.kt (hunk 2 above)
Comment (ig-garcia) [reply k9mail/src/main/java/security/planck/ui/removeaccount/RemoveAccountViewModel.kt:68]: Same as above.

File: k9mail/src/main/res/menu/menu_account_settings_option.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <!-- TODO review after adding dark theme -->
```
Comment (pzplanck) [inline k9mail/src/main/res/menu/menu_account_settings_option.xml:4]: do we have this mentioned for QA as well?

**Review by pzplanck — APPROVED** [review]

File: k9mail/src/main/java/security/planck/ui/removeaccount/RemoveAccountViewModel.kt (hunk 1 above)
Comment (Stereodark) [reply k9mail/src/main/java/security/planck/ui/removeaccount/RemoveAccountViewModel.kt:59]: where is this string displayed?
> unexpected state: ${stateLiveData.value}

does it end up in any log file? Or is it displayed in any UI?

**Pull request merged**